cookbooks/aws-parallelcluster-slurm/recipes/init_dns.rb

```diff
@@ -96,7 +96,21 @@
 # Configure fqdn in /etc/hosts
 replace_or_add "set fqdn in the /etc/hosts" do
   path "/etc/hosts"
-  pattern "^#{node['ec2']['local_ipv4']}\s+"
-  line(lazy { "#{node['ec2']['local_ipv4']} #{node['cluster']['assigned_hostname'].chomp('.')} #{node['cluster']['assigned_short_hostname']}" })
+  primary_ip = ""
+  token = get_metadata_token
+  macs = network_interface_macs(token)
+  for mac in macs
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/device-number")
+    device_number = get_metadata_with_token(token, uri)
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/network-card")
+    network_card = get_metadata_with_token(token, uri)
+    next unless device_number == '0' && network_card == '0'
+
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/local-ipv4s")
+    primary_ip = get_metadata_with_token(token, uri)
+    break
+  end
```

Comment:
I considered that while implementing but decided against it because `test_primary_ip.rb` helps to ensure that the code is correct by not reusing the functions within it.